desk/app/groups.hoon

```diff
@@ -1638,13 +1630,17 @@
         %del-sects
       ?>  go-is-bloc
       ?:  &(has-host (~(has in sects.diff) 'admin'))  go-core
-      =.  fleet.group
+      (go-fleet-del-sects ships sects.diff)
+    ==
+  ++  go-fleet-del-sects
+    |=  [ships=(set ship) sects=(set sect:g)]
+    =.  fleet.group
         %-  ~(urn by fleet.group)
         |=  [=ship =vessel:fleet:g]
         ?.  (~(has in ships) ship)  vessel
-        vessel(sects (~(dif in sects.vessel) sects.diff))
+        vessel(sects (~(dif in sects.vessel) sects))
       go-core
-    ==
+  ::
```

Comment:
I also suspect there's some affected-ships : group-size ratio for which it becomes more performant to do a couple of `+jab:by`s rather than a full `+urn:by`, but in the common case (small to medium-sized groups) this is probably faster, and always fast enough for it to not matter either way.